### PR TITLE
Fix Error Formatting Issue

### PR DESCRIPTION
**Pull request by jescalan**
So at the moment, when there is a validation error, it only outputs the stack trace, and not the actual error message. Simple reproduction:

```js
const webpack = require('webpack')
webpack({ foo: 'bar' })
```

If you run this, you'll see that the stack trace shows up, but not the actual error message, which is pretty unfortunate.

```
[...]/webpack/lib/webpack.js:19
		throw new WebpackOptionsValidationError(webpackOptionsValidationErrors);
		^

Error
    at webpack ([..]/webpack/lib/webpack.js:19:9)
    at Object.<anonymous> ([...]/test.js:3:1)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.runMain (module.js:605:10)
    at run (bootstrap_node.js:418:7)
    at startup (bootstrap_node.js:139:9)
```

I messed around with this for a while and was rather confused about why it happens, and finally stumbled across the answer. Basically, when you call `Error.captureStackTrace`, it not only captures and sets the `MyError.stack` property, it also messed with the error message a bit.

> "The first line of the trace, instead of being prefixed with ErrorType: message, will be the result of calling targetObject.toString()."
> -- [The Docs](https://nodejs.org/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt)

It does this immediately upon being called, so in this instance, it was calling `toString` on the instance **before** the error message was set, and therefore setting the error message to blank, leaving only the stack trace to show up when the error is thrown.

The fix here is simply moving it down to the bottom of the constructor, where it will run after the error message has been set. When I tested this, the error message was logged out correctly as expected 🙌

### CODE REVIEW

**Comment by timse** [comment]
can you tell me what node version you are running? 
I saw this quite a lot here in the issues but cant reproduce.
Also i think i read somewhere mentioning that you can skip the `Error.captureStackTrace` entirely as this is now default anyways, can you verify that? Just omitting these lines?

**Comment by jescalan** [comment]
Running node v7 over here. I did try removing `captureStackTrace` entirely but the result wasn't as good. The error message still contained a stack trace, but there was no `.stack` property, and the relative paths to the affected files weren't as accurate.

**Comment by timse** [comment]
can you explain: "weren't as accurate"?

**Comment by jescalan** [comment]
Sure -- they referenced the path to the file where the error occurred relative to the file throwing the error, which was in webpack/lib, instead of relative to the place that the error was ultimately thrown.

**Comment by sokra** [comment]
@jescalan Thanks for your investigation. The same construct is used very often in other Errors too. Would you like to fix these Errors too?

https://github.com/webpack/webpack/search?utf8=%E2%9C%93&q=captureStackTrace

**Comment by jescalan** [comment]
@sokra You are welcome, thanks for making webpack! It is a fantastic project, hoping I will be able to give back more as contributions in the future. I'll definitely fix this in the rest of the places I can find it.

**Comment by jescalan** [comment]
Hi! Ok so I changed over the rest of the errors in the source, they should now all be working correctly.

I also noticed that some of the errors had a check to see if `Error.captureStackTrace` was supported. According to my research, this method has been supported since at least node `0.10.x`, so I think we should be quite safe assuming it's supported with webpack2's minimum version, which, if I'm not mistaken, is `4.x`. On top of that, some errors had this check and others didn't. So I removed all these checks as well for consistency.

Let me know if there's anything else I can help out with here!

File: lib/WebpackOptionsValidationError.js [hunk 1]
```diff
@@ -51,15 +51,16 @@ class WebpackOptionsValidationError extends Error {
...
-			Error.captureStackTrace(this, this.constructor);
-		}
 		this.name = "WebpackOptionsValidationError";
 
 		this.message = "Invalid configuration object. " +
 			"Webpack has been initialised using a configuration object that does not match the API schema.\n" +
 			validationErrors.map(err => " - " + indent(WebpackOptionsValidationError.formatValidationError(err), "   ", false)).join("\n");
 		this.validationErrors = validationErrors;
+
+		if(Error.hasOwnProperty("captureStackTrace")) {
```
Comment (sokra) [inline lib/WebpackOptionsValidationError.js]: Here is one check left over.

Comment (jescalan) [inline lib/WebpackOptionsValidationError.js]: Ah good catch, not sure how I missed that one. Will remove right now 👌

**Comment by jescalan** [comment]
Ok found a couple more that slipped by the first time somehow. Want me to squash these before merge?

**Comment by timse** [comment]
it should fix https://github.com/webpack/webpack/issues/3522 as well right?

**Comment by jescalan** [comment]
It definitely seems like it to me, although the way the issue was written isn't 100% clear to me!

**Comment by sokra** [comment]
Thanks

**Pull request merged**